sklearn/model_selection/_plot.py

```diff
@@ -1,18 +1,170 @@
+
+        if std_display_style in ("fill_between", None):
+            # plot the mean score
+            if line_kw is None:
+                line_kw = {}
+
+            self.lines_ = []
+            for line_label, score in scores.items():
+                self.lines_.append(
+                    *ax.plot(
+                        x_data,
+                        score.mean(axis=1),
+                        label=line_label,
+                        **line_kw,
+                    )
+                )
+            self.errorbar_ = None
+            self.fill_between_ = None  # overwritten below by fill_between
+
+        if std_display_style == "errorbar":
+            if errorbar_kw is None:
+                errorbar_kw = {}
+
+            self.errorbar_ = []
+            for line_label, score in scores.items():
+                self.errorbar_.append(
+                    ax.errorbar(
+                        x_data,
+                        score.mean(axis=1),
+                        score.std(axis=1),
+                        label=line_label,
+                        **errorbar_kw,
+                    )
+                )
+            self.lines_, self.fill_between_ = None, None
+        elif std_display_style == "fill_between":
+            if fill_between_kw is None:
+                fill_between_kw = {}
+            default_fill_between_kw = {"alpha": 0.5}
+            fill_between_kw = {**default_fill_between_kw, **fill_between_kw}
+
+            self.fill_between_ = []
+            for line_label, score in scores.items():
+                self.fill_between_.append(
+                    ax.fill_between(
+                        x_data,
+                        score.mean(axis=1) - score.std(axis=1),
+                        score.mean(axis=1) + score.std(axis=1),
+                        **fill_between_kw,
+                    )
+                )
+
+        score_name = self.score_name if score_name is None else score_name
+
+        ax.legend()
+
+        # TODO(1.5): to be removed
+        if log_scale != "deprecated":
+            if xscale is not None:
+                raise ValueError("Cannot set both `log_scale` and `xscale`.")
+            warnings.warn(
+                (
+                    "The `log_scale` parameter is deprecated as of version 1.3 "
+                    "and will be removed in 1.5. Use `xscale` and `yscale` "
+                    "instead."
+                ),
+                FutureWarning,
+            )
+            xscale = "log" if log_scale else "linear"
+        else:
+            if xscale is None:
+                warnings.warn(
+                    (
+                        "The default value of `xscale` will change from None to 'auto'"
+                        " inversion 1.5. To avoid this warning, set `xscale` explicitly"
+                        " to'auto'."
+                    ),
+                    FutureWarning,
+                )
+                xscale = "auto"
```

Comment:
Here, we break the backward compatibility. To be strict, we would need `xscale="linear"`. However, I think the new behaviour is a UX improvement so I am wondering if we could break the backward compatibility here. We could always add in the warning above to set to `'linear'` if we want to have the previous behaviour.

ping @ogrisel @jeremiedbb

Reply:
New idea:
- deprecate "log_scale"
- don't add "xscale" and "yscale"
- the display always does the auto strategy internally
- if a user wants to change the scales, he has to get ax_ and set it himself, like all other plot customizations

Reply:
I like @jeremiedbb s proposal.

Reply:
Great I will implement this variant then.